quickwit/quickwit-lambda/src/indexer/ingest/helpers.rs

```diff
@@ -131,66 +137,116 @@ pub(super) async fn send_telemetry() {
     quickwit_telemetry::send_telemetry_event(TelemetryEvent::RunCommand).await;
 }
 
-pub(super) fn configure_source(
+/// Convert the incomming file path to a source config and save it to the metastore
+pub(super) async fn configure_source(
+    metastore: &mut MetastoreServiceClient,
     input_path: PathBuf,
     input_format: SourceInputFormat,
+    index_metadata: &IndexMetadata,
     vrl_script: Option<String>,
-) -> SourceConfig {
+) -> anyhow::Result<SourceConfig> {
     let source_params = SourceParams::file(input_path);
     let transform_config = vrl_script.map(|vrl_script| TransformConfig::new(vrl_script, None));
-    SourceConfig {
-        source_id: CLI_SOURCE_ID.to_string(),
+
+    let existing_sources_for_file: Vec<_> = index_metadata
+        .sources
+        .iter()
+        .filter(|(src_id, _)| is_lambda_source_id(src_id))
+        .map(|(src_id, src_config)| {
+            if let SourceConfig {
+                source_params: SourceParams::File(file_params),
+                ..
+            } = src_config
+            {
+                let filepath = file_params
+                    .filepath
+                    .as_ref()
+                    .context("Found Lambda file source without filepath")?;
+                Ok((src_id, filepath))
+            } else {
+                bail!("Found Lambda source with non-file source type");
+            }
+        })
+        .collect::<anyhow::Result<Vec<_>>>()?;
```

Comment:
The filter that matches the source file with the input path went missing here somehow!